docs/docs/topic-guides/custom-job-types.md

```diff
@@ -0,0 +1,90 @@
+
+### Dependencies
+
+For Python, there is a likelihood that the code to be executed has dependencies, and these can be specified one of several ways.  The docker image for the Python executor has a specific entrypoint that attempts to determine and then install the requirements.
+
+In a single file attachment (via --code) any pip commands in the module documentation are executed before the code, for instance the following code will install colorama before interpreting the script:
+
+```python
+"""
+pip install colorama 
+"""
+
+from colorama import Fore 
+```
+
+If a `pyproject.toml` is found in a directory specified by `--code`, then it is used to install the dependencies.  If the file has a poetry section, then `poetry install` is used instead.
+
+A `requirements.txt` file will be used with `pip install -r` to install requirements, if found in the specified directory.  
+
+Failing everything else, if a setup.py is present it will be used with `python -m pip install -e .`.
+
+To aid in debugging, the output of the dependency resolution is written to a file called `requirements.log` in any /outputs directory. 
+
+
+## Adding a custom job type to Bacalhau 
+ 
+To add a new experimental custom job type to Bacalhau, it is necessary to add three components. The CLI template, the requester node Transformer, and the docker image or wasm module that will execute the job on the compute node. 
+
+### CLI changes
+
+1. Add a template named after the command to be added to the [templates directory](https://github.com/bacalhau-project/bacalhau/tree/main/cmd/cli/exec/templates) (./cmd/cli/exec/templates). For example adding a template called `datafusion.tpl` will enable the command `exec datafusion`.
+2. Ensure the template contains a single task that specifies an engine that can be processed by your translator.
+3. Add any templated values you wish to be filled with full exec parameters. For example, ` "Version": "{{or (index . "version") "3.11"}}"` will add the value from `--version` or default to 3.11
+
+### Requester translator 
+
+1. Add a new implementation of [Translator](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/translation/translation.go#L30-L38) to the [translators directory](https://github.com/bacalhau-project/bacalhau/tree/main/pkg/translation/translators) in the translation package.
+2. Ensure the translation replaces the engine in the task it is provided with a new engine that is either a valid docker or webassembly engine.  
+3. Add your translator to the [standard translations provider](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/translation/translation.go#L45-L52).
```

Comment:
This implies users need to make a code change to bacalhau, correct?

Reply:
Currently, that's true.